rotkehlchen/tasks/manager.py

```diff
@@ -279,6 +287,70 @@ def _maybe_schedule_exchange_history_query(self) -> None:
         )
         self.last_exchange_query_ts[exchange.location_id()] = now
 
+    def _maybe_query_missing_prices(self) -> None:
+        query_filter = HistoryEventFilterQuery.make(limit=100)
+        entries = self.get_base_entries_missing_prices(query_filter)
+        if len(entries) > 0:
+            task_name = 'Periodically query history events prices'
+            log.debug(f'Scheduling task to {task_name}')
+            self.greenlet_manager.spawn_and_track(
+                after_seconds=None,
+                task_name=task_name,
+                exception_is_error=True,
+                method=self.query_missing_prices_of_base_entries,
+                entries_missing_prices=entries,
+            )
+
+    def get_base_entries_missing_prices(
+        self,
+        query_filter: HistoryEventFilterQuery,
+    ) -> List[Tuple[str, FVal, Asset, Timestamp]]:
+        """
+        Searches base entries missing usd prices that have not previosly been checked in
+        this session.
+        """
+        # Use a deepcopy to avoid mutations in the filter query if it is used later
+        new_query_filter = copy.deepcopy(query_filter)
+        new_query_filter.filters.append(
+            DBStringFilter(and_op=True, column='usd_value', value='0'),
+        )
+        return self.database.rows_missing_prices_in_base_entries(
+            filter_query=new_query_filter,
+            ignored_ids=self.base_entries_ignore_set,
+        )
+
+    def query_missing_prices_of_base_entries(
+        self,
+        entries_missing_prices: List[Tuple[str, FVal, Asset, Timestamp]],
+    ) -> None:
+        """Queries missing prices for HistoryBaseEntry in database updating
+        the price if it is found. Otherwise we add the id to the ignore list
+        for this session.
+        """
+        inquirer = PriceHistorian()
+        updates = []
+        for identifier, amount, asset, timestamp in entries_missing_prices:
+            try:
+                price = inquirer.query_historical_price(
+                    from_asset=asset,
+                    to_asset=A_USD,
+                    timestamp=timestamp,
+                )
+            except (NoPriceForGivenTimestamp, RemoteError) as e:
+                log.error(
+                    f'Failed to find price for {asset} at {timestamp} in base '
+                    f'entry {identifier}. {str(e)}.',
+                )
+                self.base_entries_ignore_set.add(identifier)
+                continue
+            usd_value = amount * price
```

Comment:
nitpick: This helps separate sections

```suggestion
                continue

            usd_value = amount * price
```